stats/joint_rv_types.py: Fixed broken link to reference 2 of Ewens Distribution

hanweiz opened the pull request:
#### References to other Issues or PRs
#24140 

#### Brief description of what is fixed or changed
The link of Reference 2 of `sympy.stats.MultivariateEwens` has been fixed with a redirect link.

#### Other comments


#### Release Notes

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

hanweiz committed:
stats/joint_rv_types.py: Fixed broken link to reference 2 of Ewens
Distribution

The link of Reference 2 of MultivariateEwens has been fixed with a redirect
link.

hanweiz committed:
stats/joint_rv_types.py: Fixed brooken link to reference 2 of Ewens
Distribution

The trailing whitespace in the previous commit has been removed.

hanweiz commented:
Hey @oscarbenjamin will like to check with you here if redirects are good solutions to broken links? This is the best that I can find for the reference to Ewens Distribution.

oscarbenjamin commented:
What about this link:
https://www.jstor.org/stable/24780825

hanweiz commented:
> What about this link:
> https://www.jstor.org/stable/24780825

But this requires an institutional account to read the paper no?

oscarbenjamin commented:
Someone can always google for a PDF download. What is more important with the references here is that someone can identify what document is being referred to unambiguously. It would be better if these citations actually showed the name of the paper, its authors, the journal etc rather than just having a link for a PDF.

hanweiz commented:
> Someone can always google for a PDF download. What is more important with the references here is that someone can identify what document is being referred to unambiguously. It would be better if these citations actually showed the name of the paper, its authors, the journal etc rather than just having a link for a PDF.

My observation is that these usually don't pass the `make linkcheck` that I run in `sympy/doc/` unfortunately. The same in the case for the [journal article](https://www.jstor.org/stable/24780825) you have shared, and hence my proposal of switching it to a download link.

![image](https://github.com/sympy/sympy/assets/19831350/f3702b4d-0ee5-4edc-9db1-9ff16a6d1610)

oscarbenjamin commented:
I don't know why `make linkcheck` fails but the URL is definitely valid.

The purpose of including a citation is so that in future someone can go and read that citation in order to understand something about using SymPy or about the code in SymPy. The problem with including URL is that over time they can become stale. Then if we don't have a proper citation it is difficult to know what the link previously referred to.

I would rather just have a proper citation (author name, title, journal name, ...) and not include any URL than have a citation that is only an opaque URL. If we are to include a URL then it should be a stable one that is unlikely to become stale.

hanweiz commented:
Yes a lot of the links that `make linkcheck` reports as broken are actually working valid links. Maybe I should raise an issue on it. In this case, I will amend the reference to the one you [provided](https://www.jstor.org/stable/24780825).

Seek your assistance to merge it if it passes the CI tests.

hanweiz committed:
stats/joint_rv_types: Fixed link to Reference 2 of MultivariateEwens

Replaced the link to Reference 2 of MultivariateEwens with one that points to
the original academic paper.

oscarbenjamin commented on sympy/stats/joint_rv_types.py:
```diff
@@ -633,8 +633,8 @@ def MultivariateEwens(syms, n, theta):
-    .. [2] https://www.researchgate.net/publication/280311472_The_Ubiquitous_Ewens_Sampling_Formula
-
+    .. [2] https://www.jstor.org/stable/24780825
+ 
```
This line has trailing whitespace.

oscarbenjamin replied:
I would still like the citation to list the author name, paper title etc rather than just being a URL.

smichr commented on sympy/stats/joint_rv_types.py:
```diff
@@ -633,8 +633,8 @@ def MultivariateEwens(syms, n, theta):
-    .. [2] https://www.researchgate.net/publication/280311472_The_Ubiquitous_Ewens_Sampling_Formula
-
+    .. [2] https://www.jstor.org/stable/24780825
+ 
```
```suggestion

```

smichr committed:
Pep8

smichr commented:
@hanweiz , do you want to add citation info? If not, having a working link to a stable site is an improvement already and can be committed.

hanweiz commented:
Thanks @smichr. Since this references #24140 which is specific to fixing links, I shall leave it as it is. Thanks @oscarbenjamin too.
